src/dpr/utils/sanitisedError.test.ts

```diff
@@ -0,0 +1,53 @@
+import sanitisedError, { UnsanitisedError } from './sanitisedError'
+
+describe('sanitised error', () => {
+  it('it should omit the request headers from the error object ', () => {
+    const error = {
+      name: '',
+      status: 404,
+      response: {
+        req: {
+          method: 'GET',
+          url: 'https://test-api/endpoint?active=true',
+          headers: {
+            property: 'not for logging',
+          },
+        },
+        headers: {
+          date: 'Tue, 19 May 2020 15:16:20 GMT',
+        },
+        status: 404,
+        statusText: 'Not found',
+        text: { details: 'details' },
+        body: { content: 'hello' },
+      },
+      message: 'Not Found',
+      stack: 'stack description',
+    } as unknown as UnsanitisedError
+
+    expect(sanitisedError(error)).toEqual({
+      headers: { date: 'Tue, 19 May 2020 15:16:20 GMT' },
+      message: 'Not Found',
+      stack: 'stack description',
+      status: 404,
+      text: { details: 'details' },
+      data: { content: 'hello' },
+    })
+  })
+
+  it('it should return the error message ', () => {
+    const error = {
+      message: 'error description',
+    } as unknown as UnsanitisedError
+    expect(sanitisedError(error)).toEqual({
+      message: 'error description',
+    })
+  })
+
+  it('it should return an empty object for an unknown error structure', () => {
+    const error = {
+      property: 'unknown',
+    } as unknown as UnsanitisedError
+    expect(sanitisedError(error)).toEqual({})
```

Comment:
Based on lines 25-28 of SanitisedError.ts for this test, calling the sanitisedError function with this example error, I would expect to see a response like:
```
{
message: undefined:
stack: undefined
}
```
Obviously since the test is passing, it returns an empty object but I was not sure why.

Reply:
Interestingly, if I set the expectation to the object you gave as an example, then it passes too. I think it's likely that the Jest comparison is assuming that an object with empty properties is the same as an empty object.